src/app/auth/auth.service.ts

```diff
@@ -227,10 +227,23 @@ export class AuthService {
    * @return {Observable<any>}      [description]
    */
   forgotPassword(email: string): Observable<any>  {
-    return this.request.post(api.forgotPassword, {
-      email: email,
-      domain: this.getDomain()
-    });
+    const body = {
+      email,
+      resetLink: this._createResetDirectLinks(),
+      directLink: this._createResetDirectLinks('direct')
```

Comment:
why not just separate them into two different functions.
Or make the function name more generic like `_createLinks`